File: src/cpp/power/power.cpp
unsigned long power(unsigned long x, char n)
{
    double res = 1;

    while(n>0)
    {
        if(n%2==0)
        {
            x*=x;
            n/=2;
        }
        else
        {
            res*=x;
            n--;
        }
    }

    return res;
}

File: src/js/power.js
function power(x, n)
{
    let result = 1;

    while(n > 0)
    {
        if(n % 2 == 1)
        {
            n--;
            result *= x;
        }
        else
        {
            n /= 2;
            x *= x;
        }
    }

    return result;
}